http_server.c

```diff
@@ -169,7 +169,9 @@ static int http_server_worker(void *arg)
     http_parser_init(&parser, HTTP_REQUEST);
     parser.data = &request;
     while (!kthread_should_stop()) {
-        int ret = http_server_recv(socket, buf, RECV_BUFFER_SIZE - 1);
+        int ret;
+        memset(buf, 0, RECV_BUFFER_SIZE);
+        ret = http_server_recv(socket, buf, RECV_BUFFER_SIZE - 1);
```

Comment:
just curious, could you please explain a little bit why `undefined` debug can happen in  the original version?  
```suggestion
        int ret = http_server_recv(socket, buf, RECV_BUFFER_SIZE - 1);
```

Reply:
>  just curious, could you please explain a little bit why `undefined` debug can happen in the original version?

The following code is the original version in [6808b34](https://github.com/sysprog21/khttpd/commit/6808b3415420ea3ba7f95c1c0d54df40598b3525). The variable `ret` is used at first line, but it is defined at second line. So, I rewrite the code from [6808b34](https://github.com/sysprog21/khttpd/commit/6808b3415420ea3ba7f95c1c0d54df40598b3525) to [2be19db](https://github.com/sysprog21/khttpd/commit/2be19db84e7021ed635fa8f440ddc531bf74c888)

```diff
+ memset(buf, 0, ret);
int ret = http_server_recv(socket, buf, RECV_BUFFER_SIZE - 1);
```

Reply:
> thanks for explanation. I see in [6808b34](https://github.com/sysprog21/khttpd/commit/6808b3415420ea3ba7f95c1c0d54df40598b3525), `ret` is used in `memset` , but not in new commit. So maybe the declaration and assignment of `ret` can be on the same line.

I try to declare `ret` on the same line, but compiler generates a warning
> warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
> 173 | int ret = http_server_recv(socket, buf, RECV_BUFFER_SIZE - 1);

code:
```c
memset(buf, 0, RECV_BUFFER_SIZE);
int ret = http_server_recv(socket, buf, RECV_BUFFER_SIZE - 1);
```

I'm not sure the warning will generate any problem or not. Therefore, I declare the `ret` on the first line.

Reply:
thanks for the information, I figure it out.    C99 allows for mixing declarations and statements ,but kernel still follows C90. now it looks good.

Reply:
Thanks!